fenicsxconcrete/experimental_setup/concrete_cylinder.py

```diff
@@ -0,0 +1,251 @@
+                mesh_area, self.mesh = create_cylinder_mesh(effective_radius, self.p)
+                # new guess
+                effective_radius = np.sqrt(target_area / mesh_area) * effective_radius
+
+        else:
+            raise Exception(f"wrong dimension {self.p['dim']} for problem setup")
+
+    @staticmethod
+    def default_parameters() -> dict[str, pint.Quantity]:
+        """returns a dictionary with required parameters and a set of working values as example"""
+        
+        default_parameters = {}
+        
+        # boundary setting
+        default_parameters["bc_setting"] = "free" * ureg("")  # boundary setting, two options available: fixed and free
+        # fixed: constrained at top and bottom in transversal to loading
+        # free: no confinement perpendicular to loading surface
+        
+        # mesh information
+        default_parameters["dim"] = 3 * ureg("")           # dimension of problem, 2D or 3D
+                                                           # 2D version of the cylinder is a rectangle with plane strain assumption
+        default_parameters["mesh_density"] = 4 * ureg("")  # in 3D: number of faces on the side when generating a polyhedral approximation
+                                                           # in 2D: number of elements in each direction
+        default_parameters["radius"] = 75 * ureg("mm")     # radius of cylinder to approximate in mm
+        default_parameters["height"] = 100 * ureg("mm")    # height of cylinder in mm
+        
+        return default_parameters
+
+    def create_displacement_boundary(self, V:df.fem.FunctionSpace) -> list:
+        """Defines the displacement boundary conditions
+
+        Parameters
+        ----------
+            V : Function space of the structure
+
+        Returns
+        -------
+            displ_bc : A list of DirichletBC objects, defining the boundary conditions
+        """
+
+        # define boundary conditions generator
+        bc_generator = BoundaryConditions(self.mesh, V)
+
+        if self.p['bc_setting'] == "fixed":
+            if self.p['dim'] == 2:
+                bc_generator.add_dirichlet_bc(self.top_displacement, self.boundary_top(), 1, "geometrical", 1)
+                bc_generator.add_dirichlet_bc(np.float64(0.0), self.boundary_top(), 0, "geometrical", 0)
+                bc_generator.add_dirichlet_bc(df.fem.Constant(domain=self.mesh, c=(0, 0)), self.boundary_bottom(), "geometrical")
+            elif self.p['dim']== 3:
+
+                bc_generator.add_dirichlet_bc(self.top_displacement, self.boundary_top(), 2, "geometrical", 2)
+                bc_generator.add_dirichlet_bc(np.float64(0.0), self.boundary_top(), 0, "geometrical", 0)
+                bc_generator.add_dirichlet_bc(np.float64(0.0), self.boundary_top(), 1, "geometrical", 1)
+                bc_generator.add_dirichlet_bc(df.fem.Constant(domain=self.mesh, c=(0, 0, 0)), self.boundary_bottom(), "geometrical")
```

Comment:
these lines are not tested. add a test for fixed boundaries for 2 and 3 dim please.